src/main/java/com/exadel/discount/repository/DiscountRepository.java

```diff
@@ -17,4 +17,16 @@
 
     @EntityGraph(attributePaths = {"category", "vendorLocations", "tags", "vendor"})
     Optional<Discount> findById(UUID id);
+
+    @EntityGraph(attributePaths = {"category", "vendorLocations", "tags", "vendor"})
+    List<Discount> findAllByArchivedTrue();
+
```

Comment:
you could pass archived as a parameter and reduce methods count.
` List<Discount> findAllByArchived(boolean archived)`